src/core/provider/index.js

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+const errCode = require('err-code')
+const human = require('human-to-milliseconds')
+const promisify = require('promisify-es6')
+
+const CID = require('cids')
+
+const Reprovider = require('./reprovider')
+
+class Provider {
+  /**
+   * Provider goal is to announce blocks to the network.
+   * It keeps track of which blocks are provided, and allow them to be reprovided
+   * @param {Libp2p} libp2p libp2p instance
+   * @param {Blockstore} blockstore blockstore instance
+   * @param {object} options reprovider options
+   * @param {string} options.delay reprovider initial delay in human friendly time
+   * @param {string} options.interval reprovider interval in human friendly time
+   * @param {string} options.strategy reprovider strategy
+   */
+  constructor (libp2p, blockstore, options = {}) {
+    this._running = false
+
+    this._contentRouting = libp2p.contentRouting
+    this._blockstore = blockstore
+    this._options = options
+    this.reprovider = undefined
+  }
+
+  /**
+   * Begin processing the provider work
+   * @returns {void}
+   */
+  async start () {
+    // do not run twice
+    if (this._running) {
+      return
+    }
+
+    this._running = true
+
+    // handle options (config uses uppercase)
+    const humanDelay = this._options.Delay || this._options.delay || '15s'
```

Comment:
Is `0` a valid value for the delay?

Reply:
Ok, so I think we should not accept 0, as we should wait until we are connected to some peers. Do you agree to not allow 0?

As we are using a string, that validation is very annoying 😢

Reply:
Yes I think that makes sense. The validation is a little tricky 🤔